core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -456,17 +456,25 @@ private void doStop(CompletableFuture<Void> future,
 
         private void finishDoStop(CompletableFuture<Void> future) {
             if (config.shutdownBlockingTaskExecutorOnStop()) {
+                final ExecutorService executor;
                 if (config.blockingTaskExecutor() instanceof InterminableExecutorService) {
-                    final ExecutorService service =
+                    executor =
                             ((InterminableExecutorService) config.blockingTaskExecutor()).getExecutorService();
-                    try {
-                        service.shutdown();
-                        while (!service.isTerminated()) {
-                            service.awaitTermination(500, TimeUnit.MILLISECONDS);
+                } else {
+                    executor = config.blockingTaskExecutor();
+                }
+
+                try {
+                    executor.shutdown();
+                    while (!executor.isTerminated()) {
+                        try {
+                            executor.awaitTermination(1, TimeUnit.DAYS);
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
```

Comment:
This will make the loop very busy because the next `awaitTermination` will wake up immediately and throw `InterruptedException`. Although not perfect, I'd recommend just swallowing it.